assignments/prototype-refactor.js

```diff
@@ -7,3 +7,205 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+  
+/*
+  === GameObject ===
+  * createdAt
+  * name
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+/*
+function GameObject(attrs){
+    this.createdAt = attrs.createdAt;
+    this.name = attrs.name;
+    this.dimensions = attrs.dimensions;
+  
+    this.destroy = function(){
+      return `${this.name} was removed from the game`;
+    };
+  };*/
+
+  class GameObject{
+    constructor(attrs){
+        this.createdAt = attrs.createdAt;
+        this.name = attrs.name;
+        this.dimensions = attrs.dimensions;
+    }
+
+    destroy() {
+        return `${this.name} was removed from the game`;
+    }
+  }
+  
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+
+  class CharacterStats extends GameObject{
+    constructor(attrs){
+        super(attrs);
+        this.healthPoints = attrs.healthPoints;
+    }
+
+    takeDamage() {
+        //debugger;
+        return  `${this.name} took damage.`;
+    }
+
+    destroy() {
+        return super.destroy();
+    }
+  }
+
```

Comment:
There's no need to pass down the methods when extending from a class. If the parent class has a method, anything you extend from it will receive the methods.